File: Rosella/src/main/java/me/hydos/rosella/render/material/PipelineCreateInfo.java
package me.hydos.rosella.render.material;

import me.hydos.rosella.render.Topology;
import me.hydos.rosella.render.material.state.StateInfo;
import me.hydos.rosella.render.shader.ShaderProgram;
import me.hydos.rosella.render.swapchain.RenderPass;
import me.hydos.rosella.render.vertex.VertexFormat;

public record PipelineCreateInfo(RenderPass renderPass, long descriptorSetLayout,
                                 int polygonMode, ShaderProgram shader,
                                 Topology topology,
                                 VertexFormat vertexFormat,
                                 StateInfo stateInfo) {
}

File: Rosella/src/main/kotlin/me/hydos/rosella/render/material/PipelineCreateInfo.kt
package me.hydos.rosella.render.material

import me.hydos.rosella.render.Topology
import me.hydos.rosella.render.shader.ShaderProgram
import me.hydos.rosella.render.swapchain.RenderPass
import me.hydos.rosella.render.vertex.VertexFormat

data class PipelineCreateInfo(
	val renderPass: RenderPass,
	val descriptorSetLayout: Long,
	val polygonMode: Int,
	val shader: ShaderProgram,
	val topology: Topology,
	val vertexFormat: VertexFormat,
	val useBlend: Boolean
)
